components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useState } from "react";
+import { Button } from "../../components/Button";
+import { downloadUsageCSV } from "./download-usage-csv";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const [isSaving, setIsSaving] = useState(false);
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        setIsSaving(true);
+
+        try {
+            const downloaded = await downloadUsageCSV({
+                orgName: org?.slug ?? org?.id,
+                attributionId: AttributionId.render(attributionId),
+                from: startDate.startOf("day").valueOf(),
+                to: endDate.endOf("day").valueOf(),
+            });
+
+            if (!downloaded) {
+                toast("There are no usage records for that date range");
+            }
+        } catch (e) {
+            console.error(e);
+            toast(`There was a problem downloading your usage data: ${e.message}`);
+        }
+
+        setIsSaving(false);
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" loading={isSaving} onClick={handleDownload}>
+            Download as CSV
```

Comment:
**issue(non-blocking)**: The loading state of the button when using the secondary button type is not accessible—I remember briefly discussing this with you or someone else from the `@gitpod-io/engineering-webapp` team. We could either introduce a fix for making the spinner gray or opaque when using the secondary button or leave this as is for now.

<img width="207" alt="Screenshot 2023-05-31 at 15 28 26" src="https://github.com/gitpod-io/gitpod/assets/120486/1b49b7aa-57ea-4f47-a2b7-1e1d91d58b96">